class_diagrams/core_architecture.md

```diff
@@ -0,0 +1,134 @@
+- [5]G. Young, ‘CQRS Documents by Greg Young’, p. 56.
+- [6]E. Gamma, R. Helm, R. Johnson, J. Vlissides, and G. Booch, Design Patterns: Elements of Reusable Object-Oriented Software, 1st edition. Reading, Mass: Addison-Wesley Professional, 1994.
+- [7]M. Fowler, ‘Domain Event’, martinfowler.com, Dec. 12, 2005. https://martinfowler.com/eaaDev/DomainEvent.html (accessed Dec. 05, 2020).
+- [8]E. Evans, R. Venables, and J. Fuller, Domain-Driven Design: Tackling Complexity in the Heart of Software, 1st edition. Boston: Addison-Wesley, 2003.
+- [9]G. Hohpe and B. Woolf, Enterprise Integration Patterns: Designing, Building, and Deploying Messaging Solutions, 1st edition. Boston: Addison-Wesley Professional, 2003.
+- [10]M. Fowler, ‘Event Sourcing’, martinfowler.com, Dec. 12, 2005. https://martinfowler.com/eaaDev/EventSourcing.html (accessed Dec. 05, 2020).
+- [11]T. Reenskaug, ‘MVC: XEROX PARC 1978-79’. http://heim.ifi.uio.no/~trygver/themes/mvc/mvc-index.html (accessed Dec. 05, 2020).
+- [12]B. Meyer, Object-Oriented Software Construction. New York: Prentice Hall, 1994.
+- [13]B. Meyer, Object-Oriented Software Construction, 2nd edition. Upper Saddle River, N.J: Pearson College Div, 2000.
+- [14]J. Nicola, M. Mayfield, and M. Abney, Streamlined Object Modeling: Patterns, Rules and Implementation, PAP/CDR edition. Upper Saddle River, NJ: Prentice Hall, 2001.
+- [15]L. Pacioli and  donor D. Burndy Library, Su[m]ma de arithmetica geometria proportioni [et] proportionalita. [Venice : Paganinus de Paganinis], 1494.
+- [16]J. R. Driscoll, N. Sarnak, D. D. Sleator, and R. E. Tarjan, Making Data Structures Persistent, vol. 38. Academic Press, 1989.
+
+# PlantUML
+
+```plantuml
+@startuml
+set namespaceSeparator ::
+skinparam linetype ortho
+skinparam shadowing false
+skinparam class {
+    BackgroundColor<<reducer>> HoneyDew
+    BackgroundColor<<action>> Wheat
+    BackgroundColor<<sensor>> Technology
+    BackgroundColor<<view>> Orchid
+    BackgroundColor<<effect>> Gold
+    BackgroundColor<<selector>> Lavender
+}
+
+'=========== definitions
+
+package "Core Architecture" {
+    interface Action<T> <<action>> {
+        +type:string
+        +data:T
+    }
+
+    interface Reducer <<reducer>> {
+        +reduce(state: State, action: Action): State
+    }
+
+    interface Sensor <<sensor>> {
+        +onInit():void
+        +onDestroy():void
+    }
+
+    interface Effect <<effect>> {
+        -actions$:Observable<Actions>
+    }
+
+    interface Selector <<selector>> {
+        +execute():state
+    }
+
+    interface View <<view>> {
+        +render():void
+    }
+
+    class Store <<framework>> {
+        +<<Create>> Store(reducers: Set<Reducer>)
+        -state$:Observable<State>
+        -actions$:Observable<Actions>
+        +dispatch(action: Action):void
+        +select(selector: Selector):state
+    }
+
+    class Controller <<framework>> {
+    }
```

Comment:
I think the controller contains multiple views - if so, should this be listed here - something like:

views$ ; Set<View>

Reply:
Maybe a better question - why is the controller block present only in the Core Architecture diagram ?